pages/api/auth/[...nextauth].js

```diff
@@ -1,34 +1,41 @@
 import NextAuth from "next-auth"
 import GithubProvider from "next-auth/providers/github"
-import { authenticatedPost } from "openid-client/lib/helpers/client"
 
 export default NextAuth({
     // Configure one or more authentication providers
     providers: [
-        GithubProvider({
-            clientId: process.env.GITHUB_CLIENT_ID,
-            clientSecret: process.env.GITHUB_CLIENT_SECRET,
-        }),
-        {
-          id: "artsy",
-          clientId: process.env.GRAVITY_CLIENT_ID,
-          clientSecret: process.env.GRAVITY_CLIENT_SECRET,
-          name: "Artsy",
-          type: "oauth",
-          authorization: "https://stagingapi.artsy.net/oauth2/authorize",
-          token: "https://stagingapi.artsy.net/oauth2/access_token",
-          client: {
-            token_endpoint_auth_method: "client_secret_post",
-          },
-          userinfo: "https://stagingapi.artsy.net/api/v1/me",
-          profile(profile) {
-            return {
-              id: profile.id,
-              name: profile.name,
-              email: profile.email
-            }
-          },
-        }
+      GithubProvider({
+        clientId: process.env.GITHUB_CLIENT_ID,
+        clientSecret: process.env.GITHUB_CLIENT_SECRET,
+      }),
+      {
+        id: "artsy",
+        clientId: process.env.GRAVITY_CLIENT_ID,
+        clientSecret: process.env.GRAVITY_CLIENT_SECRET,
+        name: "Artsy",
+        type: "oauth",
+        authorization: process.env.GRAVITY_AUTHORIZE_URL,
+        token: process.env.GRAVITY_TOKEN_URL,
+        client: {
+          token_endpoint_auth_method: "client_secret_post",
+        },
+        userinfo: {
+          url: process.env.GRAVITY_USERINFO_URL,
+          async request (context) {
+            const response = await fetch(context.provider.userinfo.url, {
+              headers: { "X-Access-Token": context.tokens.access_token } // override default of Authorization: Bearer ... token
```

Comment:
Does this represent another maybe-not-the-most-conformant aspect of Gravity oauth?

Reply:
Yeah I wasn't aware of that, but that's my suspicion as well.